damus/Views/PostView.swift

```diff
@@ -902,15 +902,21 @@ func build_post(state: DamusState, post: NSAttributedString, action: PostAction,
     switch action {
     case .replying_to(let replying_to):
         // start off with the reply tags
-        tags = nip10_reply_tags(replying_to: replying_to, keypair: state.keypair)
+        tags = nip10_reply_tags(replying_to: replying_to, keypair: state.keypair, relayURL: state.nostrNetwork.relaysForEvent(event: replying_to).first)
 
     case .quoting(let ev):
-        content.append("\n\nnostr:" + bech32_note_id(ev.id))
+        let relay_urls = state.nostrNetwork.relaysForEvent(event: ev)
+        if let first_relay = relay_urls.first {
+            let nevent = Bech32Object.encode(.nevent(NEvent(noteid: ev.id, relays: relay_urls.prefix(4).map { $0.absoluteString })))
+            content.append("\n\nnostr:\(nevent)")
 
-        tags.append(["q", ev.id.hex()]);
+            tags.append(["q", ev.id.hex(), first_relay.absoluteString, ev.pubkey.hex()]);
+            tags.append(["p", ev.pubkey.hex(), first_relay.absoluteString])
+        } else {
+            content.append("\n\nnostr:" + bech32_note_id(ev.id))
```

Comment:
Actually, this feels wrong for the case where we don't know which relay it came from (is this even possible? probably not, or a rare edge case when we start supporting the local relay / offline model). We can still use nevent because relays are optional but nevent also includes a pubkey. So that is strictly better than providing only a note id. I'll fix this.